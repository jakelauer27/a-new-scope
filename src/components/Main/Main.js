import React, { Component } from 'react'
import { uid } from 'react-uid'
import PropTypes from 'prop-types'
import Card from '../Card/Card'
import Loading from '../Loading/Loading'
import * as APIHelper from '../../utils/api/apiHelper'


export default class Main extends Component {
  constructor() {
    super()
    this.state = {
      categoryData: []
    }
  }

  async componentDidMount() {
    const { category, handleStoreData, changePage } = this.props
    if (category === 'favorites') {
      const favorites = JSON.parse(localStorage.getItem('favorites'))
      return this.getFavorites(favorites)
    }
    let categoryData
    const storage = JSON.parse(localStorage.getItem('storedData'))
    if (!storage || !storage[category]) {
      try {
        categoryData = await Promise.race([
          APIHelper.buildCategoryObj(category),
          new Promise((reject) => {
            setTimeout(() => reject(new Error()), 8000)
          })
        ])
        this.setState({ categoryData })
      } catch {
        return changePage('error')
      }
      return handleStoreData(category, categoryData)
    }
    return this.setState({ categoryData: storage[category] })
  }

  getFavorites(favorites) {
    if (!favorites || favorites.length === 0) {
      this.setState({
        categoryData: 'none'
      })
    } else {
      this.setState({
        categoryData: favorites
      })
    }
  }

  render() {
    const { category, changePage, handleStoreData } = this.props
    const { categoryData } = this.state
    let render

    if (!categoryData.length) {
      render = (
        <div className="App">
          <Loading />
        </div>
      )
    } else if (categoryData === 'none' || categoryData === []) {
      render = (
        <div className="no-favorite-message">You have no saved favorites</div>
      )
    } else {
      render = categoryData.map((current) => {
        return (
          <Card
            handleStoreData={handleStoreData}
            cardData={current}
            key={uid(current)}
            cardType={current.category}
          />
        )
      })
    }
    return (
      <main className="main">
        <div className="header-container">
          <div className="continue-to-site-btn back-rings" role="presentation" onClick={() => changePage('menu')}>
            <img src="./images/outerring.svg" alt="ring" className="outerring ring" />
            <img src="./images/outerring.svg" alt="ring" className="middlering ring" />
            <img src="./images/outerring.svg" alt="ring" className="innerring ring" />
          </div>
          <h1 className="main-header">{category}</h1>
        </div>
        <section className="card-section">
          {render}
        </section>
        <img
          className="return-to-landing-btn"
          alt="Landing Btn"
          src="./images/millenium_color.png"
          role="presentation"
          onClick={() => changePage('landing')}
        />
      </main>
    )
  }
}

Main.propTypes = {
  changePage: PropTypes.func.isRequired,
  category: PropTypes.string.isRequired,
  handleStoreData: PropTypes.func.isRequired
}
