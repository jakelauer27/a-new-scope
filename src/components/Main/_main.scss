.main-header,
.loading {
  color: #FCDD2B
}

.back-rings {
  position: fixed;
  top: -112px;
  left: 25px;
  cursor: pointer;
  animation: fade 2s ease;

  &:hover .ring {
    animation: rotateToSide .5s ease;
    animation-fill-mode: forwards;
    cursor: pointer;
  }
}

.header-container {
  display: flex;
  justify-content: center;
  margin-top: 30px;

  h1 {
    font-size: 5rem;
    font-family: 'Jedi';
    margin: 0;
  }

  button {
    position: absolute;
    top: 40px;
    left: 40px;
  }
}

.main {
  width: 100%;
  display: flex;
  flex-direction: column;
  overflow: scroll;
  animation: fade 2s ease;
}

.card-section {
  width: 80%;
  margin: 40px auto 50px;
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(310px, 1fr));
  grid-gap: 45px 35px;
}

.return-to-landing-btn {
  bottom: 10px;
  left: 10px;
  position: fixed;
  opacity: .5;
  transition: all .3s;
  width: 50px;

  &:hover {
    cursor: pointer;
    opacity: 1;
    transform: rotate(-45deg)
  }
}

@keyframes rotateToSide {
  100% {
    transform: rotate(190deg);
  }
}

@keyframes fade {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}