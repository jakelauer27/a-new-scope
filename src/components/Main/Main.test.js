import Main from './Main'
import { shallow } from 'enzyme'
import React from 'react';

describe('Main', () => {
  let wrapper
  let mockChangePage

  beforeEach(() => {
    mockChangePage = jest.fn()
    wrapper = shallow(<Main changePage={mockChangePage} category={'vehicles'}/>)
  })

  it('should default state to no error', () => {
    const expected = {categoryData: [], error: false}

    expect(wrapper.state()).toEqual(expected)
  })

  it('should redirect to Landing when the millenium falcon btn is clicked', () => {
    wrapper.find('.return-to-landing-btn').simulate('click')
    expect(mockChangePage).toBeCalledWith('landing')
  })

  it('should redirect to the menu page when the menu btn is clicked', () => {
    wrapper.find('.continue-to-site-btn').simulate('click')
    expect(mockChangePage).toBeCalledWith('menu')
  })

  it('Should match the snapshot when loading', () => {
    wrapper.setState({
      landingScroll: []
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('Should match the snapshot when finished loading', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('ComponentDidMount', () => {

    it('Should return category data when everything is ok', () => {

    })

    it('Should set state to error when everything is not ok', () => {

    })
  })

})
