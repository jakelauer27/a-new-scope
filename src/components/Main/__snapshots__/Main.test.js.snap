// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Should match the snapshot when finished loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    category="vehicles"
    changePage={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header-container"
        >
          <div
            className="continue-to-site-btn back-rings"
            onClick={[Function]}
          >
            <img
              alt="ring"
              className="outerring ring"
              src="./images/outerring.svg"
            />
            <img
              alt="ring"
              className="middlering ring"
              src="./images/outerring.svg"
            />
            <img
              alt="ring"
              className="innerring ring"
              src="./images/outerring.svg"
            />
          </div>
          <h1
            className="main-header"
          >
            vehicles
          </h1>
        </div>,
        <section
          className="card-section"
        >
          <div
            className="App"
          >
            <Loading />
          </div>
        </section>,
        <img
          alt="Landing Btn"
          className="return-to-landing-btn"
          onClick={[Function]}
          src="./images/millenium_color.png"
        />,
      ],
      "className": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="continue-to-site-btn back-rings"
              onClick={[Function]}
            >
              <img
                alt="ring"
                className="outerring ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="middlering ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="innerring ring"
                src="./images/outerring.svg"
              />
            </div>,
            <h1
              className="main-header"
            >
              vehicles
            </h1>,
          ],
          "className": "header-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="ring"
                  className="outerring ring"
                  src="./images/outerring.svg"
                />,
                <img
                  alt="ring"
                  className="middlering ring"
                  src="./images/outerring.svg"
                />,
                <img
                  alt="ring"
                  className="innerring ring"
                  src="./images/outerring.svg"
                />,
              ],
              "className": "continue-to-site-btn back-rings",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "ring",
                  "className": "outerring ring",
                  "src": "./images/outerring.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "ring",
                  "className": "middlering ring",
                  "src": "./images/outerring.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "ring",
                  "className": "innerring ring",
                  "src": "./images/outerring.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "vehicles",
              "className": "main-header",
            },
            "ref": null,
            "rendered": "vehicles",
            "type": "h1",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="App"
          >
            <Loading />
          </div>,
          "className": "card-section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Loading />,
            "className": "App",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Landing Btn",
          "className": "return-to-landing-btn",
          "onClick": [Function],
          "src": "./images/millenium_color.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-container"
          >
            <div
              className="continue-to-site-btn back-rings"
              onClick={[Function]}
            >
              <img
                alt="ring"
                className="outerring ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="middlering ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="innerring ring"
                src="./images/outerring.svg"
              />
            </div>
            <h1
              className="main-header"
            >
              vehicles
            </h1>
          </div>,
          <section
            className="card-section"
          >
            <div
              className="App"
            >
              <Loading />
            </div>
          </section>,
          <img
            alt="Landing Btn"
            className="return-to-landing-btn"
            onClick={[Function]}
            src="./images/millenium_color.png"
          />,
        ],
        "className": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="continue-to-site-btn back-rings"
                onClick={[Function]}
              >
                <img
                  alt="ring"
                  className="outerring ring"
                  src="./images/outerring.svg"
                />
                <img
                  alt="ring"
                  className="middlering ring"
                  src="./images/outerring.svg"
                />
                <img
                  alt="ring"
                  className="innerring ring"
                  src="./images/outerring.svg"
                />
              </div>,
              <h1
                className="main-header"
              >
                vehicles
              </h1>,
            ],
            "className": "header-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="ring"
                    className="outerring ring"
                    src="./images/outerring.svg"
                  />,
                  <img
                    alt="ring"
                    className="middlering ring"
                    src="./images/outerring.svg"
                  />,
                  <img
                    alt="ring"
                    className="innerring ring"
                    src="./images/outerring.svg"
                  />,
                ],
                "className": "continue-to-site-btn back-rings",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ring",
                    "className": "outerring ring",
                    "src": "./images/outerring.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ring",
                    "className": "middlering ring",
                    "src": "./images/outerring.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ring",
                    "className": "innerring ring",
                    "src": "./images/outerring.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "vehicles",
                "className": "main-header",
              },
              "ref": null,
              "rendered": "vehicles",
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="App"
            >
              <Loading />
            </div>,
            "className": "card-section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Loading />,
              "className": "App",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Landing Btn",
            "className": "return-to-landing-btn",
            "onClick": [Function],
            "src": "./images/millenium_color.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Main Should match the snapshot when loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    category="vehicles"
    changePage={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header-container"
        >
          <div
            className="continue-to-site-btn back-rings"
            onClick={[Function]}
          >
            <img
              alt="ring"
              className="outerring ring"
              src="./images/outerring.svg"
            />
            <img
              alt="ring"
              className="middlering ring"
              src="./images/outerring.svg"
            />
            <img
              alt="ring"
              className="innerring ring"
              src="./images/outerring.svg"
            />
          </div>
          <h1
            className="main-header"
          >
            vehicles
          </h1>
        </div>,
        <section
          className="card-section"
        >
          <div
            className="App"
          >
            <Loading />
          </div>
        </section>,
        <img
          alt="Landing Btn"
          className="return-to-landing-btn"
          onClick={[Function]}
          src="./images/millenium_color.png"
        />,
      ],
      "className": "main",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="continue-to-site-btn back-rings"
              onClick={[Function]}
            >
              <img
                alt="ring"
                className="outerring ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="middlering ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="innerring ring"
                src="./images/outerring.svg"
              />
            </div>,
            <h1
              className="main-header"
            >
              vehicles
            </h1>,
          ],
          "className": "header-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="ring"
                  className="outerring ring"
                  src="./images/outerring.svg"
                />,
                <img
                  alt="ring"
                  className="middlering ring"
                  src="./images/outerring.svg"
                />,
                <img
                  alt="ring"
                  className="innerring ring"
                  src="./images/outerring.svg"
                />,
              ],
              "className": "continue-to-site-btn back-rings",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "ring",
                  "className": "outerring ring",
                  "src": "./images/outerring.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "ring",
                  "className": "middlering ring",
                  "src": "./images/outerring.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "ring",
                  "className": "innerring ring",
                  "src": "./images/outerring.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "vehicles",
              "className": "main-header",
            },
            "ref": null,
            "rendered": "vehicles",
            "type": "h1",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="App"
          >
            <Loading />
          </div>,
          "className": "card-section",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Loading />,
            "className": "App",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Landing Btn",
          "className": "return-to-landing-btn",
          "onClick": [Function],
          "src": "./images/millenium_color.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header-container"
          >
            <div
              className="continue-to-site-btn back-rings"
              onClick={[Function]}
            >
              <img
                alt="ring"
                className="outerring ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="middlering ring"
                src="./images/outerring.svg"
              />
              <img
                alt="ring"
                className="innerring ring"
                src="./images/outerring.svg"
              />
            </div>
            <h1
              className="main-header"
            >
              vehicles
            </h1>
          </div>,
          <section
            className="card-section"
          >
            <div
              className="App"
            >
              <Loading />
            </div>
          </section>,
          <img
            alt="Landing Btn"
            className="return-to-landing-btn"
            onClick={[Function]}
            src="./images/millenium_color.png"
          />,
        ],
        "className": "main",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="continue-to-site-btn back-rings"
                onClick={[Function]}
              >
                <img
                  alt="ring"
                  className="outerring ring"
                  src="./images/outerring.svg"
                />
                <img
                  alt="ring"
                  className="middlering ring"
                  src="./images/outerring.svg"
                />
                <img
                  alt="ring"
                  className="innerring ring"
                  src="./images/outerring.svg"
                />
              </div>,
              <h1
                className="main-header"
              >
                vehicles
              </h1>,
            ],
            "className": "header-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt="ring"
                    className="outerring ring"
                    src="./images/outerring.svg"
                  />,
                  <img
                    alt="ring"
                    className="middlering ring"
                    src="./images/outerring.svg"
                  />,
                  <img
                    alt="ring"
                    className="innerring ring"
                    src="./images/outerring.svg"
                  />,
                ],
                "className": "continue-to-site-btn back-rings",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ring",
                    "className": "outerring ring",
                    "src": "./images/outerring.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ring",
                    "className": "middlering ring",
                    "src": "./images/outerring.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "ring",
                    "className": "innerring ring",
                    "src": "./images/outerring.svg",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "vehicles",
                "className": "main-header",
              },
              "ref": null,
              "rendered": "vehicles",
              "type": "h1",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="App"
            >
              <Loading />
            </div>,
            "className": "card-section",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Loading />,
              "className": "App",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Landing Btn",
            "className": "return-to-landing-btn",
            "onClick": [Function],
            "src": "./images/millenium_color.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
